sdk/core/Azure.Core/src/DynamicData/DynamicData.cs

```diff
@@ -293,8 +293,13 @@ private class JsonConverter : JsonConverter<DynamicData>
         {
             public override DynamicData Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
             {
-                using JsonDocument document = JsonDocument.ParseValue(ref reader);
-                return new DynamicData(new MutableJsonDocument(document).RootElement);
+                JsonDocument document = JsonDocument.ParseValue(ref reader);
+                using MemoryStream stream = new();
+                using Utf8JsonWriter writer = new(stream);
+                document.WriteTo(writer);
+                writer.Flush();
+                ReadOnlyMemory<byte> buffer = new(stream.GetBuffer(), 0, (int)stream.Position);
```

Comment:
This is allocating more than we need to outside of net461. Since we already multi-target Azure.Core, compare when the issue was fixed relative to what TFMs we build and do the extra code only when necessary e.g. assuming it was fixed in net47, netcoreapp3.0 and newer:

```c#
#if NET47_OR_GREATER OR NETCOREAPP3_0_OR_GREATER
// Existing code
#else
// Newer, slower code
#endif
```
```